web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateGenomeNexusController.java

```diff
@@ -74,16 +75,20 @@ public ResponseEntity<TranscriptResult> getTranscript(
         method = RequestMethod.POST)
     public ResponseEntity<List<GenomeNexusAnnotatedVariantInfo>> fetchGenomeNexusVariantInfoByHGVSgPost(
         @ApiParam(value = "List of queries. Please see swagger.json for request body format.", required = true) @RequestBody() List<AnnotateMutationByHGVSgQuery> body
-    ) throws ApiException, org.genome_nexus.ApiException, ApiHttpErrorException {
+    ) throws org.genome_nexus.ApiException, ApiHttpErrorException {
         List<GenomeNexusAnnotatedVariantInfo> result = new ArrayList<>();
 
         if (body == null) {
             throw new ApiHttpErrorException("The request body is missing.", HttpStatus.BAD_REQUEST);
         } else {
-            for (AnnotateMutationByHGVSgQuery query : body) {
-                GenomeNexusAnnotatedVariantInfo resp = GenomeNexusUtils.getAnnotatedVariantFromGenomeNexus(GNVariantAnnotationType.HGVS_G, query.getHgvsg(), query.getReferenceGenome());
-                result.add(resp);
+            ReferenceGenome referenceGenome = null;
+            if (!body.isEmpty()) {
+                referenceGenome = body.get(0).getReferenceGenome();
```

Comment:
Its not guaranteed that the reference genome is the same for the entire List<AnnotateMutationByHGVSgQuery>, but we're using the first ReferenceGenome here.

If we were to support a POST query with potentially different ReferenceGenome, then we probably have to partition the list into two, get GN annotation, then merge together?